pkg/chartrenderer/default.go

```
@@ -38,16 +37,22 @@ const notesFileSuffix = "NOTES.txt"
 // The chart render is used to render the Helm charts into a RenderedChart struct from which the
 // resulting manifest can be generated.
 type DefaultChartRenderer struct {
-	client   kubernetes.Client
-	renderer *engine.Engine
+	client       kubernetes.Client
+	renderer     *engine.Engine
+	capabilities *chartutil.Capabilities
 }
 
 // New creates a new DefaultChartRenderer object. It requires a Kubernetes client as input which will be
 // injected in the Tiller environment.
 func New(client kubernetes.Client) ChartRenderer {
+	sv, err := client.Clientset().Discovery().ServerVersion()
+	if err != nil {
+		logger.Logger.Errorf("failed to get kubernetes server version %v", err)
```

We should return an error here instead of logging and continuing.

---

I will propagate the error. However this will affect a lot of files.